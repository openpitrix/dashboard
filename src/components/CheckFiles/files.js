export default {
  vmbased: [
    {
      name: 'package.json',
      description:
        'Package.json文件，用于描述应用或版本的基本信息，包括名称版本号等'
    },
    { name: 'config.json', description: '应用的默认值配置文件' },
    { name: 'LICENSE', description: '文本格式的协议' },
    {
      name: 'locale/en.json',
      description: '应用配置的国际化英文翻译',
      isOptional: true
    },
    {
      name: 'locale/zh-en.json',
      description: '应用配置的国际化中文翻译',
      isOptional: true
    }
  ],
  helm: [
    {
      name: 'Chart.yaml',
      description: 'Yaml文件，用于描述 Chart 的基本信息，包括名称版本等'
    },
    { name: 'LICENSE', description: '文本格式的协议', isOptional: true },
    {
      name: 'README.md',
      description: '应用介绍、使用说明',
      isOptional: true
    },
    {
      name: 'requirements.yaml',
      description: '用于存放当前 Chart 依赖的其它 Chart 的说明文件',
      isOptional: true
    },
    { name: 'values.yaml', description: 'Chart 的默认值配置文件' },
    {
      name: 'charts/',
      description: '该目录中放置当前 Chart 依赖的其它 Chart',
      isOptional: true,
      check: 'none'
    },
    {
      name: 'templates/',
      description:
        '部署文件模版目录，模版填入 values.yaml 中相应值，生成最终的 kubernetes 配置文件',
      isOptional: true
    },
    { name: 'templates/NOTES.txt', description: '使用指南', isOptional: true }
  ],
  saas: [],
  api: [],
  native: [],
  serveless: []
};
