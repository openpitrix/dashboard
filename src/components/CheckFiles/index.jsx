import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import classnames from 'classnames';
import _ from 'lodash';

import { versionTypes } from 'config/version-types';
import typeFiles from './files';

import styles from './index.scss';

@translate()
export default class CheckFiles extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    errorFiles: PropTypes.array,
    isShowNote: PropTypes.bool,
    type: PropTypes.string,
    uploadStatus: PropTypes.string
  };

  static defaultProps = {
    type: 'vmbased',
    errorFiles: [],
    isShowNote: false,
    uploadStatus: ''
  };

  render() {
    const {
      className,
      type,
      errorFiles,
      isShowNote,
      uploadStatus,
      t
    } = this.props;
    const files = typeFiles[type] || [];
    const linkType = (_.find(versionTypes, { value: type }) || {}).name;

    return (
      <div className={classnames(styles.checkFiles, className)}>
        <ul>
          {files.map(file => (
            <li
              key={file.name}
              className={classnames({
                [styles.error]: errorFiles.includes(file.name)
              })}
            >
              <span className={styles.name}>{file.name}</span>
              <div className={styles.description}>
                #&nbsp;&nbsp;
                {file.isOptional && <span>[{t('optional')}]&nbsp;&nbsp;</span>}
                {t(file.description)}
              </div>
            </li>
          ))}
        </ul>
        {uploadStatus === 'init' && <div className={styles.configMask} />}
        {isShowNote && (
          <div className={styles.document}>
            <span className={styles.note}>{t('Note')}</span>
            {t('See the complete app development specification')}
            <a to="#">
              《{linkType}
              {t('specification and application development')}
              》
            </a>。
          </div>
        )}
      </div>
    );
  }
}
