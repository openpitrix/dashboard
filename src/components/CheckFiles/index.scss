@import "~scss/vars";

.checkFiles {
  position: relative;

  li {
    margin-bottom: 8px;
    padding: 6px 20px;
    line-height: 20px;
    background-color: $N0;
    font-size: 12px;
    text-align: left;

    &.none {
      opacity: 0.5;
    }

    &.error {
      .name, .description {
        color: $R200;
      }
    }
  }

  .name {
    float: left;
    display: inline-block;
    width: 25%;
    color: $N300;
  }

  .description {
    margin-left: 25%;
    width: 75%;
    color: $N75;
  }

  .configMask {
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    z-index: 2;
    background-image: linear-gradient(to bottom, rgba($body-bg, 0.75), $body-bg);
  }

  .document {
    margin-top: 20px;
    line-height: 20px;
    font-size: 12px;
    color: $N65;

    .note {
      float: left;
      margin-right: 8px;
      display: inline-block;
      width: 32px;
      height: 20px;
      border-radius: 1px;
      background-color: $N30;
      color: $N100;
      text-align: center;
    }
  }
}
