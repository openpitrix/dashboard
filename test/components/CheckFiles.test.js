import CheckFiles from 'components/CheckFiles';

describe('Component CheckFiles', () => {
  it('basic render', () => {
    const wrapper = render(
      <CheckFiles className='test' uploadStatus="init" isShowNote />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
