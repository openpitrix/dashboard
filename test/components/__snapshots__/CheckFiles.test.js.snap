// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CheckFiles basic render 1`] = `
<div
  class="checkFiles test"
>
  <ul>
    <li
      class=""
    >
      <span
        class="name"
      >
        package.json
      </span>
      <div
        class="description"
      >
        #  PACKAGE_FILE_DESC
      </div>
    </li>
    <li
      class=""
    >
      <span
        class="name"
      >
        config.json
      </span>
      <div
        class="description"
      >
        #  CONFIG_FILE_DESC
      </div>
    </li>
    <li
      class=""
    >
      <span
        class="name"
      >
        LICENSE
      </span>
      <div
        class="description"
      >
        #  LICENSE_FILE_DESC
      </div>
    </li>
    <li
      class=""
    >
      <span
        class="name"
      >
        locale/en.json
      </span>
      <div
        class="description"
      >
        #  
        <span>
          [optional]  
        </span>
        LOCALE_EN__FILE_DESC
      </div>
    </li>
    <li
      class=""
    >
      <span
        class="name"
      >
        locale/zh-en.json
      </span>
      <div
        class="description"
      >
        #  
        <span>
          [optional]  
        </span>
        LOCALE_ZH_FILE_DESC
      </div>
    </li>
  </ul>
  <div
    class="configMask"
  />
  <div
    class="document"
  >
    <span
      class="note"
    >
      Note
    </span>
    See the complete app development specification
    <a
      href="https://docs.openpitrix.io/v0.3/zh-CN/developer-guide/openpitrix-developer-guide/"
      target="blank"
    >
      《SPECIFICATION_LINK_TITLE》
    </a>
    。
  </div>
</div>
`;
